harvard-lts/hdc3a-transfer-service
# Allow for empty data directory which requires no hash mapping file

**Allow empty data directory and no hash mapping file*
* * *

# How should this be tested?
Make sure the unit test passes:

1. Clone the repo
2. Checkout the branch blank_data
3. Copy the env-template.txt to .env
4. Get the credentials from LPE Shared-HDataCommons/3a/dims ActiveMQ dev/qa transfer and process queue for ims-transfer-dev.  Add the credentials to the .env
6. Get the credentials from LPE Shared-HDataCommons/3a/dataverse-export-s3 dev/qa for the dataverse-export-dev access key and secret access key and set them in the .env as appropriate.
7. In the .env, set` SUPPLIED_HASH_MAPPING_FILENAME=manifest-md5.txt`
8. In the .env, set` REQUIRED_UNZIPPED_FILES=bag-info.txt,bagit.txt,metadata/datacite.xml,metadata/oai-ore.jsonld,metadata/pid-mapping.txt`
9. Start up the docker container: `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
10. Exec into the container: `docker exec -it hdc3a-transfer-service bash`
11. Run the command `tests/unit/test_transfer_service.py`
